### PR TITLE
`sage.{calculus,functions,numerical,symbolic}`: Docstring/doctest cosmetics, `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
Cherry-picked from #35095.

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by DaveWitteMorris** [comment]
I'm sorry, but I can't review this, because I don't understand modularization.

File: src/sage/functions/gamma.py [hunk 1]
```diff
@@ -721,7 +721,8 @@ def gamma(a, *args, **kwds):
         sage: gamma(i)                                                                  # needs sage.rings.number_field sage.symbolic
         Traceback (most recent call last):
         ...
-        TypeError: cannot coerce arguments: no canonical coercion from Number Field in i with defining polynomial x^2 + 1 to Symbolic Ring
+        TypeError: cannot coerce arguments: no canonical coercion
+        from Number Field in i with defining polynomial x^2 + 1 to Symbolic Ring
```
Comment (kwankyu) [inline src/sage/functions/gamma.py:726]: ` from` (a space before `from`)?

File: src/sage/symbolic/expression.pyx [hunk 2]
```diff
@@ -12890,7 +12890,8 @@ cdef class Expression(Expression_abc):
             sage: plot(2*sin, -4, 4)                                                    # needs sage.plot
             Traceback (most recent call last):
             ...
-            TypeError: unsupported operand parent(s) for *: 'Integer Ring' and '<class 'sage.functions.trig.Function_sin'>'
+            TypeError: unsupported operand parent(s) for *:
+            'Integer Ring' and '<class 'sage.functions.trig.Function_sin'>'
```
Comment (kwankyu) [inline src/sage/symbolic/expression.pyx:12895]: What is the principle here?

I think, if a long line (the output) is split to multiple lines, the second and the following lines should be indented by a space. No?

**Comment by kwankyu** [comment]
> I'm sorry, but I can't review this, because I don't understand modularization.

@mkoeppe For this business of maintaining the "# needs" or "# optional" tags, I think that it should be obvious to a PR author how to modify or add or delete those tags. I guess that "Test modularized distributions" of "B&T" would (could) annotate "Files changed" with appropriate messages on how to modify them. If I am right, is it working now or are you working for it to work?

**Comment by kwankyu** [comment]
Otherwise, lgtm.

File: src/sage/symbolic/expression.pyx (hunk 2 above)
Comment (mkoeppe) [reply src/sage/symbolic/expression.pyx:12895]: I don't do that in error messages

**Comment by mkoeppe** [comment]
> For this business of maintaining the "# needs" or "# optional" tags, I think that it should be obvious to a PR author how to modify or add or delete those tags. I guess that "Test modularized distributions" of "B&T" would (could) annotate "Files changed" with appropriate messages on how to modify them. If I am right, is it working now or are you working for it to work?

This can only be done after the distributions in #35095 have been merged into `develop`. 
In the current branch, these tags just cannot be tested yet.

File: src/sage/symbolic/expression.pyx (hunk 2 above)
Comment (kwankyu) [reply src/sage/symbolic/expression.pyx:12895]: OK.

**Comment by kwankyu** [comment]
I see. Thanks.

**Review by kwankyu — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thank you!

**Comment by vbraun** [comment]
fail, see CI

**Pull request merged**